config/src/iroha.rs

```diff
@@ -60,29 +63,76 @@ view! {
     }
 }
 
-impl Default for Configuration {
+impl Default for ConfigurationProxy {
     fn default() -> Self {
-        let sumeragi_configuration = sumeragi::Configuration::default();
-        let (public_key, private_key) = sumeragi_configuration.key_pair.clone().into();
-
         Self {
-            public_key,
-            private_key,
-            disable_panic_terminal_colors: bool::default(),
-            kura: kura::Configuration::default(),
-            sumeragi: sumeragi_configuration,
-            torii: torii::Configuration::default(),
-            block_sync: block_sync::Configuration::default(),
-            queue: queue::Configuration::default(),
-            logger: logger::Configuration::default(),
-            genesis: genesis::Configuration::default(),
-            wsv: wsv::Configuration::default(),
-            network: network::Configuration::default(),
-            telemetry: telemetry::Configuration::default(),
+            public_key: None,
+            private_key: None,
+            disable_panic_terminal_colors: Some(bool::default()),
+            kura: Some(kura::ConfigurationProxy::default()),
+            sumeragi: Some(sumeragi::ConfigurationProxy::default()),
+            torii: Some(torii::ConfigurationProxy::default()),
+            block_sync: Some(block_sync::ConfigurationProxy::default()),
+            queue: Some(queue::ConfigurationProxy::default()),
+            logger: Some(logger::ConfigurationProxy::default()),
+            genesis: Some(genesis::ConfigurationProxy::default()),
+            wsv: Some(wsv::ConfigurationProxy::default()),
+            network: Some(network::ConfigurationProxy::default()),
+            telemetry: Some(telemetry::ConfigurationProxy::default()),
         }
     }
 }
 
+impl ConfigurationProxy {
+    pub fn finalize(&mut self) -> Result<()> {
+        if let Some(sumeragi_proxy) = &mut self.sumeragi {
+            // First, iroha public/private key and sumeragi keypair are interchangeable, but at least
```

Comment:
I would suggest to prohibit providing keys to sumeragi at all (it would be sumeragi private field). Imo there should be only one way to configurate keys.

Reply:
Makes sense. I wanted to leave it more flexible, but that actually complicates finalization and such, so your approach should allow for a leaner implementation.

Reply:
>  I wanted to leave it more flexible, but that actually complicates finalization and such

Correct assessment. But I'd be more specific. Configs need to have certain fields in sync. The easiest (and most correct) way to achieve that is by enforcing that synchronised config types are created together. 

You can prevent accidental desynchronised configs from being created by hiding necessary fields which should be synced. This is what @Erigara is proposing, which IMO is the better approach.

Reply:
Opted for just overwriting whatever was in the `key_pair` field for now. Would that be sufficient or maybe it'll be more prudent to also slap a `#[serde(skip)]` on the relevant proxy field? It would be a bit hacky as that has to be done inside the derive macro itself.

Reply:
I think overwriting is ok for now, but in the future i think we should exclude such fields from de/serialization, because there is no point in having fields that will be overwritten in any case.

Reply:
Agree, making an issue: #2715.